projects/make-a-vis/src/lib/legend-view/graphic-variable-legend/graphic-variable-legend.component.ts

```diff
@@ -52,50 +50,15 @@ export class GraphicVariableLegendComponent implements OnInit, OnChanges {
     const template = this.graphicVariableOption.visualization;
     this.legendVisualizationType = template;
     if (!!template && graphicVariable) {
-      const preData: any = {
-        id: `legend-visualization-${uniqueId()}`,
-        template,
-        properties: {},
-        graphicSymbols: {}
-      };
-      this.store.pipe(
-        select(getLoadedProjectSelector)
-      ).subscribe(project => {
-        this.serializer.createVisualization(
-          this.graphicVariableOption.visualization, preData, project
-        ).subscribe((legend) => {
-          legend.graphicSymbols.items = this.generateLegendGraphicSymbol(template, graphicVariable, graphicSymbol, project);
-          this.legend = legend;
-          this.legendComponent.data = legend;
-          this.legendComponent.runDataChangeDetection();
-        });
+      this.legendService.createLegend(template, graphicVariable, graphicSymbol).subscribe((legend) => {
+      this.legend = legend;
```

Comment:
You should move these three lines out to a private function so you don't have to repeat the oddness here and in the else clause below.